main/tests/server/src/com/google/refine/expr/functions/arrays/JoinTests.java

```diff
@@ -28,13 +28,35 @@
 
 import org.testng.annotations.Test;
 
+import com.google.refine.RefineTest;
+import com.google.refine.expr.ParsingException;
 import com.google.refine.util.TestUtils;
 
-public class JoinTests {
+public class JoinTests extends RefineTest {
     @Test
     public void serializeJoin() {
         String json = "{\"description\":\"Returns the string obtained by joining the array a with the separator sep\",\"params\":\"array a, string sep\",\"returns\":\"string\"}";
         TestUtils.isSerializedTo(new Join(), json);
     }
+
+    @Test
+    public void joinArray() throws ParsingException {
+        String[] test = {"[2,1,3].join('|')", "2|1|3"};
+        parseEval(bindings, test);
+
+        // TODO: This is current behavior, but is it what we want?
+        String[] test1 = {"[2,null,3].join(', ')", "2, 3"};
+        parseEval(bindings, test1);
+
+        String[] test2 = {"['z','b','c','a'].join('-')", "z-b-c-a"};
+        parseEval(bindings, test2);
+
+        // TODO: Do we really want the following two tests to return different results?
```

Comment:
Yes, they should return differently.  I think that was agreed in #820 and it's linked issues/pr's